nmrfx-core/src/main/java/org/nmrfx/project/ProjectBase.java

```diff
@@ -145,11 +147,10 @@ List<DatasetBase> getDatasetList() {
 
     public List<DatasetBase> getDatasetsWithFile(File file) {
         try {
-            String testPath = file.getCanonicalPath();
-            List<DatasetBase> datasetsWithFile = datasetMap.values().stream().filter((dataset) -> (dataset.getCanonicalFile().equals(testPath))).collect(Collectors.toList());
-            return datasetsWithFile;
+            String testPath = requireNonNullElse(file.getCanonicalPath(), "");
+            return datasetMap.values().stream().filter(dataset -> (testPath.equals(dataset.getCanonicalFile()))).collect(Collectors.toList());
```

Comment:
Since I used the in memory file, getCanonicalFile was returning null and causing an exception, changed to default to an empty string. I'm not sure if I should have tried comparing the dataset names instead if they are memory files?